Xplat/src/main/java/vazkii/botania/common/entity/ManaSparkEntity.java

```diff
@@ -362,4 +368,44 @@ public boolean areIncomingTransfersDone() {
 		return attachable != null && attachable.areIncomingTranfersDone();
 	}
 
+	public record WandHud(ManaSparkEntity entity) implements WandHUD {
+		@Override
+		public void renderHUD(PoseStack ms, Minecraft mc) {
+			final ItemStack sparkItem = new ItemStack(BotaniaItems.spark);
+
+			int color = 0x4444FF;
+			{
+				final Component sparkName = sparkItem.getHoverName();
+				int width = mc.font.width(sparkName) / 2;
+				int x = mc.getWindow().getGuiScaledWidth() / 2 - width;
+				int y = mc.getWindow().getGuiScaledHeight() / 2 + 5;
+
+				mc.font.drawShadow(ms, sparkName, x, y + 5, color);
+			}
+
+			{
+				final DyeColor network = this.entity.getNetwork();
+				final Component colorName = Component.translatable("color.minecraft." + network.getName()).withStyle(ChatFormatting.ITALIC);
+				int width = mc.font.width(colorName) / 2;
+				int x = mc.getWindow().getGuiScaledWidth() / 2 - width;
+				int y = mc.getWindow().getGuiScaledHeight() / 2 + 20;
+
+				mc.font.drawShadow(ms, colorName, x, y + 5, network.getTextColor());
+			}
+
+			final SparkUpgradeType upgrade = this.entity.getUpgrade();
+			if (upgrade != SparkUpgradeType.NONE) {
+				final ItemStack upgradeItem = SparkAugmentItem.getByType(upgrade);
+				final Component upgradeName = upgradeItem.getHoverName();
+				int width = 16 + mc.font.width(upgradeName) / 2;
+				int x = mc.getWindow().getGuiScaledWidth() / 2 - width;
+				int y = mc.getWindow().getGuiScaledHeight() / 2 + 50;
+
+				mc.font.drawShadow(ms, upgradeName, x + 20, y + 5, color);
+				mc.getItemRenderer().renderAndDecorateItem(upgradeItem, x, y);
+			}
+
+			RenderSystem.setShaderColor(1F, 1F, 1F, 1F);
```

Comment:
is this needed? iirc none of the methods we're calling here should clobber the global color

Reply:
no it's not, oops